src/chatd.cpp

```diff
@@ -4144,4 +4178,97 @@ bool Message::isValidEmail(const string &buf)
     return regex_match(buf, regularExpresion);
 }
 
+FilteredHistory::FilteredHistory(DbInterface *db)
+    : mDb(db)
+{
+    initVaraiables();
+    mDb->getHistoryNodeIndex(mNewest, mOldest);
+}
+
+void FilteredHistory::addMessage(const Message &msg, bool isNew)
+{
+    Message* message = new Message(msg);
+    if (isNew)
+    {
+        mBuffer.emplace_front(message);
+        mNewest ++;
+        mDb->addAttachmentMessageToHistoryNode(msg, mNewest);
+    }
+    else
+    {
+        if (find(msg.id()) == mBuffer.end())
```

Comment:
This operation (insert old attachments retrieved from server) may become too heavy with a long history, since for every message we need to go through the whole list. Would it be possible to assume the message is not part of the node-history yet and add it always?

Reply:
Solved it